xds/internal/balancer/clusterresolver/resource_resolver_dns.go

```diff
@@ -142,7 +142,17 @@ func (dr *dnsDiscoveryMechanism) ReportError(err error) {
 		dr.logger.Infof("DNS discovery mechanism for resource %q reported error: %v", dr.target, err)
 	}
 
-	dr.topLevelResolver.onError(err)
+	dr.mu.Lock()
+	// Don't report any errors after first error or good address update.
+	if dr.addrs != nil {
```

Comment:
Should this be `if !dr.updateReceived { ... }` instead?

Reply:
Yeah. It can be. I was debating whether using the addrs or the bool. I guess if you don't like the addrs:= make(resolver.Address, 0) call I can use the bool.